tt_metal/tools/tt_builder/main.cpp

```diff
@@ -0,0 +1,71 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <iostream>
+#include "builder.hpp"
+#include "tt_metal/common/logger.hpp"
+using namespace std;
+
+void print_usage(const char* exec_name) {
+    cout << "Usage: " << exec_name << " [OPTION]" << endl;
+    cout << "\t-h, --help           : Display this message." << endl;
+    cout << "\t--build-firmware     : Build Firmware." << endl;
+    cout << "\t--build-kernel       : Build Kernel." << endl;
```

Comment:
let's remove "build-kernel" and add "build-dispatch".  This will get used for building internal kernels.  I can't think of how we can use this utility to build user kernels, I think we'll just have to provide the infra for them to create their own "builder" program that builds their kernels